### PR TITLE
Integrate with RTEMS

### PR DESCRIPTION
**Pull request by lounick**
This PR provides integration with RTEMS real-time OS. It provides the required files for RTEMS integration plus minor fixes and refactoring to reduce code duplication (e.g. the slaveinfo example).

### CODE REVIEW

**Comment by nakarlsson** [comment]
Thank you for the pull request,  Please review the CLA to be able to contribute to SOEM, Link to CLA can be found in the README.md at the bottom. We'll start to review when the CLA is signed.

**Comment by lounick** [comment]
@nakarlsson any news on this? I sent the document like 10 days ago. As I see I have to remove or change some of the Travis stuff so that it is aligned to your settings.

**Comment by nakarlsson** [comment]
Yes, it’s in progress.

File: test/linux/slaveinfo/slaveinfo.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/** \file
+ * \brief Declaration of slaveinfo for SOEM.
```
Comment (hefloryd) [inline test/linux/slaveinfo/slaveinfo.h]: Could you motivate why this is being extracted? Do you intend to use it outside of SOEM? Currently this file is not referenced anywhere in the tree as far as I can tell. It should probably at least be included by slaveinfo.c

File: test/linux/slaveinfo/slaveinfo.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+#ifndef TEST_LINUX_SLAVEINFO_SLAVEINFO_H_
```
Comment (hefloryd) [inline test/linux/slaveinfo/slaveinfo.h]: SLAVEINFO_H should be enough? Would avoid having to rename the guard if the file is moved for any reason.

File: test/linux/slaveinfo/slaveinfo.h (hunk 1 above)
Comment (lounick) [reply test/linux/slaveinfo/slaveinfo.h]: I was running some tests with RTEMS. Possibly I should clean up those and push them as well as a separate RTEMS tests.

**Comment by lounick** [comment]
Ok so on monday I will clean up the travis things. Try to integrate also the slaveinfo I had for RTEMS and rebase to squash the commits. Then you can take a final look and merge. Thanks for the review!  :smile:

**Comment by hefloryd** [comment]
Merged, thanks!

**Comment by lounick** [comment]
No worries! Happy to contribute! :smile:

**Pull request merged**